### DISCUSSION TITLE
DuckDB crash when preparing too long sql

### DISCUSSION THREAD

**Discussion by yxmm-wxe** [duckdb/duckdb#2571] [Q&A] [answered]
When the DuckDB executes a too-long SQL containing more than 3000 CASE WHEN statements, DuckDB will crash. 
My Java program process is also finished.

```
Process finished with exit code -1073741571 (0xC00000FD)
```

According to my research, the error code represents that maximum recursion depth exceeded.
And the issue happened in [DuckDBNative.duckdb_jdbc_prepare](https://github.com/duckdb/duckdb/blob/master/tools/jdbc/src/main/java/org/duckdb/DuckDBPreparedStatement.java#L73)

I'd like to know the limitation of the SQL in DuckDB, and if the issue is a bug?

**Comment by Mytherin** [comment, marked as answer]
Indeed this is likely related to recursion depth, but a crash should never happen, so this is indeed a bug. Please file a bug report!
